tests/foreman/api/test_hostgroup.py

```diff
@@ -148,6 +148,64 @@ def test_bz_1107708(self):
         self.assertEqual(host_attrs['all_puppetclasses'][0]['name'], 'ntp')
 
 
+@skip_if_bug_open('bugzilla', 1235377)
+class MissingAttrTestCase(APITestCase):
+    """Tests to see if the server returns the attributes it should."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a ``HostGroup`` as ``cls.host_group``."""
+        cls.host_group = entities.HostGroup().create()
+
+    def test_get_content_source(self):
+        """@Test: Read a host group. Inspect the server's response.
+
+        @Assert: The response contains either a ``content_source`` or
+        ``content_source_id`` attribute.
+
+        @Feature: HostGroup
+
+        """
+        wanted_attrs = ('content_source', 'content_source_id')
```

Comment:
Instead of hard-coding these attribute names, I could simply call [`nailgun.entity_mixins._get_entity_id`](http://nailgun.readthedocs.org/en/latest/api/nailgun.entity_mixins.html#nailgun.entity_mixins._get_entity_id) instead. This has a couple advantages:

* There's no need to write out each permutation of an attribute name.
* `_get_entity_id()` is already capable of throwing an exception containing a nice error message when the desired attribute is not found.

It would look something like this:

```python
_get_entity_id('content_source', self.host_group.read_json())
```

The down-side here is that I'm calling a private function. However, the benefits are significant, so I'm thinking that it's worthwhile. Thoughts?